build/devenv/tests/e2e/finality_reorg_test.go

```diff
@@ -0,0 +1,190 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/rs/zerolog"
+	"github.com/stretchr/testify/require"
+
+	"github.com/smartcontractkit/chainlink-ccip/ccv/chains/evm/deployment/v1_7_0/operations/committee_verifier"
+	"github.com/smartcontractkit/chainlink-ccip/ccv/chains/evm/deployment/v1_7_0/operations/executor"
+	ccv "github.com/smartcontractkit/chainlink-ccv/devenv"
+	cciptestinterfaces "github.com/smartcontractkit/chainlink-ccv/devenv/cciptestinterfaces"
+	"github.com/smartcontractkit/chainlink-ccv/devenv/evm"
+	"github.com/smartcontractkit/chainlink-ccv/protocol"
+	"github.com/smartcontractkit/chainlink-deployments-framework/datastore"
+	"github.com/smartcontractkit/chainlink-testing-framework/framework"
+)
+
+// TestSimpleReorgWithMessageOrdering tests that messages sent in different orders
+// before and after a reorg are correctly verified after finality is reached.
+// IMPORTANT: Need to run this test against an env that has source chain with auto mining.
+// Run `just rebuild-all "env.toml,env-src-auto-mine.toml"` before running this test.
+func TestSimpleReorgWithMessageOrdering(t *testing.T) {
+	in, err := ccv.LoadOutput[ccv.Cfg]("../../env-out.toml")
+	require.NoError(t, err)
+
+	ctx := ccv.Plog.WithContext(context.Background())
+	l := zerolog.Ctx(ctx)
+
+	chainIDs, wsURLs := make([]string, 0), make([]string, 0)
+	for _, bc := range in.Blockchains {
+		chainIDs = append(chainIDs, bc.ChainID)
+		wsURLs = append(wsURLs, bc.Out.Nodes[0].ExternalWSUrl)
+	}
+
+	selectors, e, err := ccv.NewCLDFOperationsEnvironment(in.Blockchains, in.CLDF.DataStore)
+	require.NoError(t, err)
+	require.Len(t, selectors, 3, "expected 3 chains for this test in the environment")
+
+	c, err := evm.NewCCIP17EVM(ctx, *l, e, chainIDs, wsURLs)
+	require.NoError(t, err)
+
+	t.Cleanup(func() {
+		_, err := framework.SaveContainerLogs(fmt.Sprintf("%s-%s", framework.DefaultCTFLogsDir, t.Name()))
+		require.NoError(t, err)
+	})
+
+	defaultAggregatorAddr := fmt.Sprintf("127.0.0.1:%d", defaultAggregatorPort(in))
+
+	defaultAggregatorClient, err := ccv.NewAggregatorClient(
+		zerolog.Ctx(ctx).With().Str("component", "aggregator-client").Logger(),
+		defaultAggregatorAddr)
+	require.NoError(t, err)
+	require.NotNil(t, defaultAggregatorClient)
+	t.Cleanup(func() {
+		defaultAggregatorClient.Close()
+	})
+
+	// Get the source and destination chain selectors
+	srcSelector := selectors[0]
+	destSelector := selectors[1]
```

Comment:
Do we need to ensure that certain blockchain properties are valid on these selectors? e.g. that `srcSelector` points to `blockchain-src` in the env-src-auto-mine.toml?